DataFormats/Portable/interface/AlpakaCollection.h

```diff
@@ -44,6 +46,14 @@ class AlpakaCollection {
 
   Buffer const &buffer() const { return *buffer_; }
 
+#if defined ALPAKA_ACC_CPU_B_SEQ_T_SEQ_ENABLED || 1
+  template <typename T2>
+  static void ROOTReadStreamer(AlpakaCollection * newObj, T2 onfile) {
+    new (newObj) AlpakaCollection(onfile.layout_.size(), host);
```

Comment:
I just realised that with this approach, the destructor of the original `AlpakaCollection` object at `newObj` is never called.
While the `T` in the alpaka collection is likely to have a trivial destructor, either we check it with _e.g._ `if constexpr(std::is_trivially_destructible_v<T>)`, or we should just call the destructor and let the ccmpiler elide the trivial cases:
```c++
     newObj->~AlpakaCollection();
     new (newObj) AlpakaCollection(onfile.layout_.size(), host);
```